homeassistant/components/unifi/sensor.py

```diff
@@ -18,51 +22,23 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up sensors for UniFi integration."""
     controller = get_controller_from_config_entry(hass, config_entry)
-    sensors = {}
-
-    option_allow_bandwidth_sensors = controller.option_allow_bandwidth_sensors
-
-    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    controller.entities[DOMAIN] = {RX_SENSOR: set(), TX_SENSOR: set()}
 
     @callback
     def items_added():
         """Update the values of the controller."""
-        nonlocal option_allow_bandwidth_sensors
-
-        if not option_allow_bandwidth_sensors:
-            return
-
-        add_entities(controller, async_add_entities, sensors)
+        if controller.option_allow_bandwidth_sensors:
+            add_entities(controller, async_add_entities)
 
     controller.listeners.append(
         async_dispatcher_connect(hass, controller.signal_update, items_added)
     )
 
-    @callback
-    def items_removed(mac_addresses: set) -> None:
-        """Items have been removed from the controller."""
-        remove_entities(controller, mac_addresses, sensors, entity_registry)
-
-    controller.listeners.append(
-        async_dispatcher_connect(hass, controller.signal_remove, items_removed)
-    )
-
     @callback
     def options_updated():
         """Update the values of the controller."""
-        nonlocal option_allow_bandwidth_sensors
-
-        if option_allow_bandwidth_sensors != controller.option_allow_bandwidth_sensors:
-            option_allow_bandwidth_sensors = controller.option_allow_bandwidth_sensors
-
-            if option_allow_bandwidth_sensors:
-                items_added()
-
-            else:
-                for sensor in sensors.values():
-                    hass.async_create_task(sensor.async_remove())
-
-                sensors.clear()
+        if controller.option_allow_bandwidth_sensors:
```

Comment:
Looks like we can drop this check. It's done inside `items_added` too.

Reply:
Maybe we can even drop the `options_updated` callback and use `items_added` directly as callback?

Reply:
Yeah I realised that as well cleaning it all up